src/Compilers/CSharp/Portable/Parser/LanguageParser_Patterns.cs

```diff
@@ -680,7 +680,9 @@ private SeparatedSyntaxList<SwitchExpressionArmSyntax> ParseSwitchExpressionArms
                 arms.Add(switchExpressionCase);
                 if (this.CurrentToken.Kind != SyntaxKind.CloseBraceToken)
                 {
-                    var commaToken = this.EatToken(SyntaxKind.CommaToken);
+                    var commaToken = this.CurrentToken.Kind == SyntaxKind.SemicolonToken
+                        ? this.EatTokenAsKind(SyntaxKind.CommaToken)
+                        : this.EatToken(SyntaxKind.CommaToken);
```

Comment:
👍 

One thing to consider is *always* doing this.  We went htat approach in TS: https://github.com/microsoft/TypeScript/blob/b24050aefd2a3ea88d463c6a147a9b6e1878dec7/src/compiler/parser.ts#L2117-L2128

Or, another way of putting it: we just had a generalized list parsing algorithm, and then it was trivial to examine each call to parse a list and ask "should we just pass in the bool here since users could reasonably be expected to pass the wrong separator kind in".

Anyways, something to maybe consider in the future :)